### PR TITLE
Integration test - restore from local group snapshots.

### PR DESCRIPTION
**Pull request by Rohit-PX**
Signed-off-by: Rohit-PX <rohit@portworx.com>

**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
This is a new integration test to restore group volume snapshots into PVCs

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
2.1

### CODE REVIEW

**Comment by Rohit-PX** [comment]
Execution logs: location(Master node) 70.0.53.37 :/root/stest.log

File: test/integration_test/snapshot_test.go [hunk 1]
```diff
@@ -13,17 +13,22 @@ import (
...
 var snapRuleFailRegex = regexp.MustCompile("^snapshot failed due to err.+(failed to validate snap rule|failed to run (pre|post)-snap rule).+")
+var storkStorageClass = "stork-snapshot-sc"
 
 func testSnapshot(t *testing.T) {
 	t.Run("simpleSnapshotTest", simpleSnapshotTest)
 	t.Run("cloudSnapshotTest", cloudSnapshotTest)
 	t.Run("snapshotScaleTest", snapshotScaleTest)
 	t.Run("groupSnapshotTest", groupSnapshotTest)
 	t.Run("groupSnapshotScaleTest", groupSnapshotScaleTest)
+	t.Run("groupSnapshotRestoreTest", groupSnapshotRestoreTest)
```
Comment (disrani-px) [inline test/integration_test/snapshot_test.go]: Make this a part of `groupSnapshotTest`, that is already testing creation of snaps

File: test/integration_test/snapshot_test.go [hunk 2]
```diff
@@ -166,6 +171,62 @@ func groupSnapshotScaleTest(t *testing.T) {
...
+
+	for _, ctx := range ctxs {
+		verifyGroupSnapshot(t, ctx, groupSnapshotWaitTimeout)
+	}
+
+	for _, ctx := range ctxs {
+		err := schedulerDriver.WaitForRunning(ctx, defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for pod to get to running state")
+
+		snaps, err := schedulerDriver.GetSnapshots(ctx)
```
Comment (disrani-px) [inline test/integration_test/snapshot_test.go]: Check error and verify length of snapshots

File: test/integration_test/snapshot_test.go [hunk 3]
```diff
@@ -166,6 +171,62 @@ func groupSnapshotScaleTest(t *testing.T) {
...
+		logrus.Infof("Number of Snapshot: %d", len(snaps))
+		for _, snap := range snaps {
+			err = createRestorePvcForSnap(snap.Name, snap.Namespace)
```
Comment (disrani-px) [inline test/integration_test/snapshot_test.go]: You also need to verify that the PVC gets bound. We want to make sure it doesn't fail.

File: test/integration_test/snapshot_test.go [hunk 4]
```diff
@@ -166,6 +171,62 @@ func groupSnapshotScaleTest(t *testing.T) {
...
+		}
+	}
+	ctxsToDestroy = append(ctxsToDestroy, ctxs...)
+
+	destroyAndWait(t, ctxsToDestroy)
+}
+
+func getSnapAnnotation(snapName string) map[string]string {
+	snapAnnotation := make(map[string]string)
+	snapAnnotation["snapshot.alpha.kubernetes.io/snapshot"] = snapName
```
Comment (disrani-px) [inline test/integration_test/snapshot_test.go]: The annotation is available from this package: `github.com/kubernetes-incubator/external-storage/snapshot/pkg/client`

File: test/integration_test/snapshot_test.go [hunk 5]
```diff
@@ -134,11 +144,39 @@ func groupSnapshotTest(t *testing.T) {
...
+		snaps, err := schedulerDriver.GetSnapshots(ctx)
+		require.NoError(t, err, fmt.Sprintf("Failed to get snapshots for %s.", ctx.App.Key))
+		numSnaps := len(snaps)
+		require.Lenf(t, numSnaps, 2, fmt.Sprintf("Only %d snapshots created for %s expected 2.", len(snaps), ctx.App.Key))
+		for _, snap := range snaps {
+			restoredPvc, err := createRestorePvcForSnap(snap.Name, snap.Namespace)
+			require.NoError(t, err, fmt.Sprintf("Failed to create pvc for restoring snapshot %s.", snap.Name))
+
+			err = k8s.Instance().ValidatePersistentVolumeClaim(restoredPvc, waitPvcBound, waitPvcRetryInterval)
+			require.NoError(t, err, fmt.Sprintf("PVC for restored snapshot %s not bound.", snap.Name))
```
Comment (disrani-px) [inline test/integration_test/snapshot_test.go]: Delete the PVC after verifying

File: test/integration_test/snapshot_test.go [hunk 6]
```diff
@@ -9,21 +9,31 @@ import (
...
+const (
+	waitPvcBound         = 120 * time.Second
+	waitPvcRetryInterval = 5 * time.Second
+)
 
 func testSnapshot(t *testing.T) {
-	t.Run("simpleSnapshotTest", simpleSnapshotTest)
-	t.Run("cloudSnapshotTest", cloudSnapshotTest)
-	t.Run("snapshotScaleTest", snapshotScaleTest)
+	//t.Run("simpleSnapshotTest", simpleSnapshotTest)
```
Comment (disrani-px) [inline test/integration_test/snapshot_test.go]: Uncomment the tests

File: test/integration_test/specs/mysql-groupsnap-restore/group_snap.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+apiVersion: stork.libopenstorage.org/v1alpha1
```
Comment (disrani-px) [inline test/integration_test/specs/mysql-groupsnap-restore/group_snap.yaml]: These aren't being used anymore, right?

Comment (Rohit-PX) [reply test/integration_test/specs/mysql-groupsnap-restore/group_snap.yaml]: Deleted both spec files.

**Comment by disrani-px** [comment]
Also, please squash into one commit

File: test/integration_test/snapshot_test.go (hunk 1 above)
Comment (Rohit-PX) [reply test/integration_test/snapshot_test.go]: Fixed.

File: test/integration_test/snapshot_test.go (hunk 2 above)
Comment (Rohit-PX) [reply test/integration_test/snapshot_test.go]: Fixed.

File: test/integration_test/snapshot_test.go (hunk 3 above)
Comment (Rohit-PX) [reply test/integration_test/snapshot_test.go]: Fixed.

File: test/integration_test/snapshot_test.go (hunk 4 above)
Comment (Rohit-PX) [reply test/integration_test/snapshot_test.go]: Done.

File: test/integration_test/snapshot_test.go (hunk 5 above)
Comment (Rohit-PX) [reply test/integration_test/snapshot_test.go]: Deleting PVC.

File: test/integration_test/snapshot_test.go (hunk 6 above)
Comment (Rohit-PX) [reply test/integration_test/snapshot_test.go]: Done.

**Pull request merged**